lib/core/include/irods/escape_utilities.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef IRODS_ESCAPE_UTILITIES_HPP
+#define IRODS_ESCAPE_UTILITIES_HPP
+
+#include <string>
+
+namespace irods
+{
+    /// Returns a copy of a string with all single quotes replaced by their hexadecimal value.
+    ///
+    /// \param[in] _s A string possibly containing single quotes.
+    ///
+    /// \returns std::string
+    ///
+    /// \since 5.0.0
+    auto single_quotes_to_hex(const std::string& _s) -> std::string;
```

Comment:
Maybe mention that this `throw`s an `irods::exception`.

Reply:
Updated.